### PR TITLE
Check hash against delivered collection file

### PR DESCRIPTION
**Pull request by benoitdevos**
* Check hash against delivered collection file.
* Re-enable expand button in collection item files.

logion-network/logion-internal#738
logion-network/logion-internal#758

### CODE REVIEW

**Review by gdethier — APPROVED** [review]
Approved,  but I think some adjustments are needed.

File: src/certificate/Certificate.tsx [hunk 1]
```diff
@@ -27,7 +37,7 @@ import LegalOfficerRow from "./LegalOfficerRow";
 import { Children } from "src/common/types/Helpers";
 import CheckDeliveredFrame from "src/components/deliverycheck/CheckDeliveredFrame";
 import ClaimAssetButton, { walletType } from "./ClaimAssetButton";
-
+import { CheckCertifiedCopyResult } from "@logion/client/dist/CollectionItem.js";
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply src/certificate/Certificate.tsx:40]: Done.

File: src/certificate/Certificate.tsx [hunk 2]
```diff
@@ -150,6 +160,32 @@ export default function Certificate() {
...
+        if (isItemFileDelivery(collectionItem)) {
+            const result = await collectionItem!.checkCertifiedCopy(hash);
+            if (result.summary === CheckResultType.POSITIVE) {
+                return result;
+            }
+        }
+        const result = await loc.checkCertifiedCopy(hash);
+        if (result.summary === CheckResultType.POSITIVE) {
+            const file = loc.data.files.find(file => file.hash === result.match?.originalFileHash);
+            setCheckResult({ file })
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply src/certificate/Certificate.tsx:182]: Yes. This surrounds the file with dotted square.

File: src/certificate/Certificate.tsx [hunk 3]
```diff
@@ -150,6 +160,32 @@ export default function Certificate() {
...
+        } else {
+            setCheckResult(undefined);
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply src/certificate/Certificate.tsx:184]: This one is needed also, but needs to be moved to erase the square when needed.

File: src/certificate/Certificate.tsx [hunk 4]
```diff
@@ -150,6 +160,32 @@ export default function Certificate() {
         return result;
     }
 
+    function hasRestrictedDeliveryCheckTool(loc: PublicLoc, collectionItem: CollectionItem | null | undefined): boolean {
+        const isCollectionFileDelivery = loc.data.files.find(file => file.restrictedDelivery) !== undefined;
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply src/certificate/Certificate.tsx:164]: Indeed. The sole presence of files within the LOC, regardless of their restricted delivery status, should trigger the display of the check box.

File: src/components/itemfiles/ItemFiles.tsx [hunk 5]
```diff
@@ -130,15 +130,16 @@ export default function ItemFiles(props: Props) {
     );
 }
 
-function numberOfClaims(file: DeliveredFile, deliveries?: ItemDeliveriesResponse): string {
-    if(deliveries !== undefined) {
+function numberOfClaims(file: DeliveredFile, deliveries?: ItemDeliveriesResponse): { value: string, hideExpand: boolean } {
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply src/components/itemfiles/ItemFiles.tsx:133]: Done in this [commit](https://github.com/logion-network/logion-wallet/pull/424/commits/9e2b71ccd2ea2c6b4e499c4af8aedcb13c4727cd).

**Pull request merged**